libsql-client/src/workers.rs

```diff
@@ -0,0 +1,170 @@
+use async_trait::async_trait;
+use base64::Engine;
+use worker::*;
+
+use super::{parse_query_result, QueryResult, Statement};
+
+/// Database connection. This is the main structure used to
+/// communicate with the database.
+#[derive(Clone, Debug)]
+pub struct Connection {
+    url: String,
+    auth: String,
+}
+
+impl Connection {
+    /// Establishes a database connection.
+    ///
+    /// # Arguments
+    /// * `url` - URL of the database endpoint
+    /// * `username` - database username
+    /// * `pass` - user's password
+    pub fn connect(
+        url: impl Into<String>,
+        username: impl Into<String>,
+        pass: impl Into<String>,
+    ) -> Self {
+        let username = username.into();
+        let pass = pass.into();
+        let url = url.into();
+        // Auto-update the URL to start with https:// if no protocol was specified
+        let url = if !url.contains("://") {
+            "https://".to_owned() + &url
+        } else {
+            url
```

Comment:
Would be cool to change `http://` prefix to `https://` automatically too.

Reply:
@honzasp explicitly asked not to do it in one of the past reviews and I agreed with the sentiment - turso.io indeed uses https://, but technically users could run their own local testing servers that run on http://

Reply:
Fair enough. Maybe at least a warning would be good though.